doc/classes/Array.xml

```diff
@@ -183,36 +174,46 @@
 			<return type="void" />
 			<param index="0" name="value" type="Variant" />
 			<description>
-				Appends an element at the end of the array (alias of [method push_back]).
+				Appends [param value] at the end of the array (alias of [method push_back]).
 			</description>
 		</method>
 		<method name="append_array">
 			<return type="void" />
 			<param index="0" name="array" type="Array" />
 			<description>
-				Appends another array at the end of this array.
+				Appends another [param array] at the end of this array.
 				[codeblock]
-				var array1 = [1, 2, 3]
-				var array2 = [4, 5, 6]
-				array1.append_array(array2)
-				print(array1) # Prints [1, 2, 3, 4, 5, 6].
+				var numbers = [1, 2, 3]
+				var extra = [4, 5, 6]
+				numbers.append_array(extra)
+				print(nums) # Prints [1, 2, 3, 4, 5, 6]
 				[/codeblock]
 			</description>
 		</method>
 		<method name="back" qualifiers="const">
 			<return type="Variant" />
 			<description>
-				Returns the last element of the array. Prints an error and returns [code]null[/code] if the array is empty.
-				[b]Note:[/b] Calling this function is not the same as writing [code]array[-1][/code]. If the array is empty, accessing by index will pause project execution when running from the editor.
+				?? Returns the last element of the array. Generates an error and returns [code]null[/code] if the array is empty.
+				?? Returns the last element of the array. If the array is empty, generates an error and returns [code]null[/code], without stopping the project execution (unlike with the [code][][/code] operator).
+				?? [b]Note:[/b] This method is not the same as [code]array[-1][/code].
 			</description>
 		</method>
 		<method name="bsearch">
 			<return type="int" />
 			<param index="0" name="value" type="Variant" />
 			<param index="1" name="before" type="bool" default="true" />
 			<description>
-				Finds the index of an existing value (or the insertion index that maintains sorting order, if the value is not yet present in the array) using binary search. Optionally, a [param before] specifier can be passed. If [code]false[/code], the returned index comes after all existing entries of the value in the array.
-				[b]Note:[/b] Calling [method bsearch] on an unsorted array results in unexpected behavior.
+				Returns the index of [param value] in the sorted array. If it cannot be found, returns where [param value] should be inserted to keep the array sorted. The algorithm used is [url=https://en.wikipedia.org/wiki/Binary_search_algorithm]binary search[/url].
+				?? If [param before] is [code]false[/code], the returned index comes after all existing entries of the [param value] in the array.
+				?? If [param before] is [code]true[/code] (as by default), the returned index comes before all elements that are equal to [param value] in the array.
```

Comment:
It's a bit weird to say "before is false", and then say "after". May be just me, I just am not sure.

Reply:
Only minor problem is, I actively avoided calling them entries on purpose. It's not an incorrect term, it was just a matter of consistency in making sure it's always "element".